core/src/test/scala/org/apache/spark/deploy/history/ApplicationCacheSuite.scala

```diff
@@ -381,4 +385,38 @@ class ApplicationCacheSuite extends SparkFunSuite with MockitoSugar with Matcher
     verify(resp).sendRedirect("http://localhost:18080/history/local-123/jobs/job/?id=2")
   }
 
+  test("Load new SparkUI during old one is detaching") {
```

Comment:
The new case should named as `SPARK-43403: ... xxx`.

And I think it would be best if the new case could be reproduced  `java.lang.IllegalStateException: DB is closed` without change of `ApplicationCache` and pass with this pr.

If it is not easy to reproduce `DB is closed` (without change of ApplicationCache) in this one, I think you can add other `test`

Reply:
OK. Let me try.